src/renderer/page/settings/view.jsx

```diff
@@ -1,352 +1,321 @@
-      this.setState({ clearingCache: false });
-      window.location.href = 'index.html';
-    };
-    const clear = () => this.props.clearCache().then(success.bind(this));
 
-    setTimeout(clear, 1000, { once: true });
+    (this: any).onDownloadDirChange = this.onDownloadDirChange.bind(this);
+    (this: any).onKeyFeeChange = this.onKeyFeeChange.bind(this);
+    (this: any).onInstantPurchaseMaxChange = this.onInstantPurchaseMaxChange.bind(this);
+    (this: any).onShowNsfwChange = this.onShowNsfwChange.bind(this);
+    (this: any).onShowUnavailableChange = this.onShowUnavailableChange.bind(this);
+    (this: any).onShareDataChange = this.onShareDataChange.bind(this);
+    (this: any).onThemeChange = this.onThemeChange.bind(this);
+    (this: any).onAutomaticDarkModeChange = this.onAutomaticDarkModeChange.bind(this);
+    (this: any).clearCache = this.clearCache.bind(this);
+    // (this: any).onLanguageChange = this.onLanguageChange.bind(this)
   }
 
-  setDaemonSetting(name, value) {
-    this.props.setDaemonSetting(name, value);
+  componentDidMount() {
+    this.props.getThemes();
   }
 
-  onRunOnStartChange(event) {
+  onRunOnStartChange(event: SyntheticInputEvent<*>) {
     this.setDaemonSetting('run_on_startup', event.target.checked);
   }
 
-  onShareDataChange(event) {
+  onShareDataChange(event: SyntheticInputEvent<*>) {
     this.setDaemonSetting('share_usage_data', event.target.checked);
   }
 
-  onDownloadDirChange(event) {
-    this.setDaemonSetting('download_directory', event.target.value);
+  onDownloadDirChange(newDirectory: string) {
+    this.setDaemonSetting('download_directory', newDirectory);
   }
 
-  onKeyFeeChange(newValue) {
-    const setting = newValue;
-
-    // this is stupid and should be fixed... somewhere
-    if (setting && (setting.amount === undefined || setting.amount === null)) {
-      setting.amount = 0;
-    }
-
-    this.setDaemonSetting('max_key_fee', setting);
+  onKeyFeeChange(newValue: Price) {
+    this.setDaemonSetting('max_key_fee', newValue);
   }
 
-  onKeyFeeDisableChange(isDisabled) {
+  onKeyFeeDisableChange(isDisabled: boolean) {
     this.setDaemonSetting('disable_max_key_fee', isDisabled);
   }
 
-  onThemeChange(event) {
+  onThemeChange(event: SyntheticInputEvent<*>) {
     const { value } = event.target;
     this.props.setClientSetting(settings.THEME, value);
   }
 
-  onAutomaticDarkModeChange(event) {
+  onAutomaticDarkModeChange(event: SyntheticInputEvent<*>) {
     this.props.setClientSetting(settings.AUTOMATIC_DARK_MODE_ENABLED, event.target.checked);
   }
 
-  onInstantPurchaseEnabledChange(enabled) {
+  onInstantPurchaseEnabledChange(enabled: boolean) {
     this.props.setClientSetting(settings.INSTANT_PURCHASE_ENABLED, enabled);
   }
 
-  onInstantPurchaseMaxChange(newValue) {
+  onInstantPurchaseMaxChange(newValue: Price) {
     this.props.setClientSetting(settings.INSTANT_PURCHASE_MAX, newValue);
   }
 
-  // onMaxUploadPrefChange(isLimited) {
```

Comment:
Figured we can remove these. They aren't to complex so adding them back should be easy (when we can).